File: 5_Recursion-10mar/C3-12mar/subsequenceOfString/incToExc.js
function printSubString(s, output, i) {
  if (i >= s.length) {
    console.log(output)
    return
  }
  // include 
  output = output + s[i]
  printSubString(s, output, i + 1)
  //exclude
  output = output.slice(0, output.length - 1)//? slice exclude the 2arg index
  printSubString(s, output, i + 1)
}

const s = "abc"
let output = ""
let i = 0
printSubString(s, output, i)

File: 5_Recursion-10mar/C3-12mar/subsequenceOfString/incToexc.cpp
#include <iostream>
#include <vector>
using namespace std;

void printSubString(string s, string output, int i)
{
    if (i >= s.length())
    {
        
        cout << output << endl;
        return;
    }
    // include
    output.push_back(s[i]);
    printSubString(s, output, i + 1);
    output.pop_back();

    // exclude
    printSubString(s, output, i + 1);
}

int main()
{
    string s = "abc";
    string output = "";
    int i = 0;
    printSubString(s, output, i);

    // ans ? cout << "sorted arr" << endl : cout << "not " << endl;

    return 0;
}